.github/copilot-instructions.md

```diff
@@ -72,58 +72,266 @@ const [state, dispatch] = useReducer(modalReducer, initialState)
 // ✅ Good: Context for global state
 const { user, setUser } = useContext(UserContext)
 
-// ✅ Good: React Query for server state
+// ✅ Good: React Query for client-side server state (when route loaders aren't suitable)
 const { data: users, isLoading } = useQuery({
   queryKey: ['users'],
   queryFn: fetchUsers,
 })
 ```
 
+### React Query vs Route Loaders
+
+**Prefer Route Loaders for:**
+- Initial page data that's required for rendering
+- Server-side rendering (SSR) requirements
+- Data that should be fetched before the component mounts
+- Critical path data needed for SEO
+
+```typescript
+// ✅ Good: Use loader for initial page data
+export const Route = createFileRoute('/users')({
+  loader: async (): Promise<{ users: User[] }> => {
+    const users = await fetchUsers();
+    return { users: userListSchema.parse(users) };
+  },
+  component: UserList,
+});
+```
+
+**Prefer React Query for:**
+- Data that updates frequently
+- Optional/secondary data not critical for initial render
+- Client-side mutations and optimistic updates
+- Data that needs fine-grained cache control
+
+```typescript
+// ✅ Good: Use React Query for dynamic/optional data
+function UserStats({ userId }: { userId: string }) {
+  const { data: stats, isLoading } = useQuery({
+    queryKey: ['user-stats', userId],
+    queryFn: () => fetchUserStats(userId),
+    refetchInterval: 30000, // Auto-refresh every 30s
+  });
+
+  if (isLoading) return <div>Loading stats...</div>;
+  return <div>{stats?.totalPosts} posts</div>;
+}
+```
+
 ## Validation with Zod
 
-Always define schemas in `src/lib/schemas.ts` and reuse them:
+**Always validate external data** - from APIs, user input, environment variables, and any data crossing system boundaries.
+
+### Schema Organization and Best Practices
+
+Define all schemas in `src/lib/schemas.ts` with proper documentation:
```

Comment:
Currently all schemas are in `src/lib/schemas.ts` but probably it could be improved to have longterm a `src/lib/schemas` folder? I started off with `src/lib/schemas.ts` because there wasn't many.